plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1010,6 +1022,8 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
 
         Set<Integer> partitionSourceIds = identityPartitionColumnsInAllSpecs(icebergTable);
         BiPredicate<IcebergColumnHandle, Domain> isIdentityPartition = (column, domain) -> partitionSourceIds.contains(column.getId());
+        // Iceberg metadata columns can not be used in table scans
+        BiPredicate<IcebergColumnHandle, Domain> isMetadataColumn = (column, domain) -> Arrays.stream(values()).anyMatch(metadataColumn -> metadataColumn.getId() == column.getId());
```

Comment:
> values()

It would be better to avoid unqualified method name because we can't understand the source of the class when reading. We could add a helper method in `IcebergMetadataColumn` likes this:
```java
    FILE_PATH(MetadataColumns.FILE_PATH.fieldId(), "$path", VARCHAR, PRIMITIVE),
    /**/;

    private static final Set<Integer> COLUMNS_ID = Stream.of(values())
            .map(IcebergMetadataColumn::getId)
            .collect(toImmutableSet());
...

    public static boolean isMetadataColumn(int id)
    {
        return COLUMNS_ID.contains(id);
    }
```

Reply:
@ebyhr sure, can add another method. Initially thought about that, but then as it can be done in a single line, so chose the later option.

Reply:
updated @ebyhr